### ISSUE TITLE
Include a flutter preview device with the Flutter SDK that supports hot reload

### ISSUE THREAD

**Issue by christopherfujino** [flutter/flutter#130277]
### Use case

This would allow new Flutter developers to experience Flutter's hot reload developer experience before installing a native build toolchain (such as Xcode, or Visual Studio).

- [x] introduce a new, hidden `flutter build _preview` sub-command https://github.com/flutter/flutter/pull/135639
- [x] introduce a new, PreviewDevice device available to Windows host users on release channels https://github.com/flutter/flutter/pull/135639
- [x] add error handling for when a user tries to build an app with plugins https://github.com/flutter/flutter/pull/136936
- [x] ensure IDE extensions can detect preview device https://github.com/flutter/flutter/pull/140112
- [ ] Have VSCode list the device as "preview -preview device", not referring to desktop
- [ ] Figure out why the preview device does not show in the "Flutter sidebar" in VSCode
- [ ] distribute the preview device with the Flutter SDK
- [ ] https://github.com/flutter/flutter/issues/139949

**Comment by andrewkolos** [comment]
@christopherfujino Is this a dupe of https://github.com/flutter/flutter/issues/85329?

**Comment by christopherfujino** [comment]
> @christopherfujino Is this a dupe of #85329?

good catch, closed that in favor of this one, as it more closely reflects the steps I'm working on.
